src/tree/hist/evaluate_splits.h

```diff
@@ -119,13 +119,90 @@ class HistEvaluator {
     p_best->Update(best);
   }
 
+  /**
+   * \brief Enumerate with partition-based splits.
+   *
+   * The implementation is different from LightGBM. Firstly we don't have a
+   * pseudo-cateogry for missing value, instead of we make 2 complete scans over the
+   * histogram. Secondly, both scan directions generate splits in the same
+   * order. Following table depicts the scan process, square bracket means the gradient in
+   * missing values is resided on that partition:
+   *
+   *   | Forward  | Backward |
+   *   |----------+----------|
+   *   | A [BCDE] | [ABCD] E |
+   *   | AB [CDE] | [ABC] DE |
+   *   | ABC [DE] | [AB] CDE |
+   *   | ABCD [E] | [A] BCDE |
+   */
+  template <int d_step>
```

Comment:
Add comment: in this function, d_step=+1 indicates the backward direction, and d_step=-1 indicates the forward direction.

Reply:
Actually +1 is the forward, it's just a convention from one hot split that the gradient is being added to the right node, which might be confusing.

Reply:
In that case, switch the two columns in the example table.

Reply:
Done.